ReportingServicesTools/Functions/CatalogItems/New-RsFolder.ps1

```diff
@@ -89,30 +88,35 @@ function New-RsFolder
     $namespace = $proxy.GetType().Namespace
     $propertyDataType = "$namespace.Property"
     $additionalProperties = New-Object System.Collections.Generic.List[$propertyDataType]
-    if ($Description)
-    {
+    if ($Description) {
         $descriptionProperty = New-Object $propertyDataType
         $descriptionProperty.Name = 'Description'
         $descriptionProperty.Value = $Description
         $additionalProperties.Add($descriptionProperty)
     }
 
-    if ($Hidden)
-    {
+    if ($Hidden) {
         $hiddenProperty = New-Object $propertyDataType
         $hiddenProperty.Name = 'Hidden'
         $hiddenProperty.Value = $Hidden
         $additionalProperties.Add($hiddenProperty)
     }
 
-    try
-    {
+    try {
         Write-Verbose "Creating folder $($FolderName)..."
-        $Proxy.CreateFolder($FolderName, $RsFolder, $additionalProperties) | Out-Null
-        Write-Verbose "Folder $($FolderName) created successfully!"
+        $checkRsFolder = $RsFolder
+        if ($checkRsFolder[-1] -ne '/') {
+            $checkRsFolder += '/'
+        }
+        if ($Proxy.GetItemType("$($checkRsFolder)$FolderName") -ne "Folder") {
```

Comment:
it is the caller responsibility to catch the exceptions and/or validate the folder exists before calling this function

Check for example the pshell command New-Item with ItemType "directory" which shares the same behavior of failure when there is a existing folder

Reply:
Not sure why I didn't think of that lol. Thanks!